airflow-core/src/airflow/task/priority_strategy.py

```diff
@@ -41,46 +42,22 @@ class PriorityWeightStrategy(ABC):
     """
 
     @abstractmethod
-    def get_weight(self, ti: TaskInstance):
+    def get_weight(self, ti: TaskInstance) -> int:
         """Get the priority weight of a task."""
-        ...
-
-    @classmethod
-    def deserialize(cls, data: dict[str, Any]) -> PriorityWeightStrategy:
-        """
-        Deserialize a priority weight strategy from data.
-
-        This is called when a serialized DAG is deserialized. ``data`` will be whatever
-        was returned by ``serialize`` during DAG serialization. The default
-        implementation constructs the priority weight strategy without any arguments.
-        """
-        return cls(**data)
-
-    def serialize(self) -> dict[str, Any]:
-        """
-        Serialize the priority weight strategy for JSON encoding.
-
-        This is called during DAG serialization to store priority weight strategy information
-        in the database. This should return a JSON-serializable dict that will be fed into
-        ``deserialize`` when the DAG is deserialized. The default implementation returns
-        an empty dict.
-        """
-        return {}
+        raise NotImplementedError("must be implemented by a subclass")
```

Comment:
We probably should add a newsfragment, update any related docs too

Reply:
I added a note for this.